Subject: [PATCH] Fix potential uninitialized variable uses

Initialize CurrentCertLen in X509GetCertFromCertChain and ensure FfaEnabled
is read before the loop in DelegatedEventLoop to avoid potential use of
uninitialized values.

diff --git i/CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptX509.c w/CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptX509.c
index f0727135ad..db9920853d 100644
--- i/CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptX509.c
+++ w/CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptX509.c
@@ -857,6 +857,7 @@ X509GetCertFromCertChain (
 
   CurrentCert  = CertChain;
   CurrentIndex = -1;
+  CurrentCertLen = 0;
 
   //
   // Traverse the certificate chain
diff --git i/StandaloneMmPkg/Library/StandaloneMmCoreEntryPoint/Arm/StandaloneMmCoreEntryPoint.c w/StandaloneMmPkg/Library/StandaloneMmCoreEntryPoint/Arm/StandaloneMmCoreEntryPoint.c
index 70306be249..446f47bec1 100644
--- i/StandaloneMmPkg/Library/StandaloneMmCoreEntryPoint/Arm/StandaloneMmCoreEntryPoint.c
+++ w/StandaloneMmPkg/Library/StandaloneMmCoreEntryPoint/Arm/StandaloneMmCoreEntryPoint.c
@@ -128,6 +128,8 @@ DelegatedEventLoop (
   EFI_STATUS  Status;
   UINTN       SvcStatus;
 
+  FfaEnabled = FeaturePcdGet (PcdFfaEnable);
+
   while (TRUE) {
     ArmCallSvc (EventCompleteSvcArgs);
 
@@ -152,7 +154,6 @@ DelegatedEventLoop (
       DEBUG ((DEBUG_ERROR, "UnRecognized Event - 0x%x\n", (UINT32)EventCompleteSvcArgs->Arg0));
       Status = EFI_INVALID_PARAMETER;
     } else {
-      FfaEnabled = FeaturePcdGet (PcdFfaEnable);
       if (FfaEnabled) {
         Status = CpuDriverEntryPoint (
                    EventCompleteSvcArgs->Arg0,
